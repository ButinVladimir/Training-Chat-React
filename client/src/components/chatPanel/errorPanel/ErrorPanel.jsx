import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

function ErrorPanel({ error, onCloseError }) {
  return (
    <div className="error-panel-container">
      <div className="error-panel">
        <div className="popup-row">An error occurred</div>
        <div className="popup-row">{ error }</div>
        <div className="popup-row">
          <button className="close-button" type="button" onClick={onCloseError}>Close</button>
        </div>
      </div>
    </div>
  );
}

ErrorPanel.propTypes = {
  error: PropTypes.string,
  onCloseError: PropTypes.func.isRequired,
};

ErrorPanel.defaultProps = {
  error: '',
};

export default ErrorPanel;
