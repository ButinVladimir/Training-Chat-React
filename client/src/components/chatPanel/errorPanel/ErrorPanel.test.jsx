import React from 'react';
import { shallow } from 'enzyme';
import ErrorPanel from './ErrorPanel';

describe('ErrorPanel', () => {
  it('renders without crashing', () => {
    const component = shallow(<ErrorPanel />);
    expect(component).toMatchSnapshot();
  });
});
