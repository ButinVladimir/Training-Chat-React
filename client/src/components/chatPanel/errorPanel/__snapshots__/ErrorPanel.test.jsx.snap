// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPanel renders without crashing 1`] = `
<div
  className="error-panel-container"
>
  <div
    className="error-panel"
  >
    <div
      className="popup-row"
    >
      An error occurred
    </div>
    <div
      className="popup-row"
    />
    <div
      className="popup-row"
    >
      <button
        className="close-button"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
